### PR TITLE
Compare error handling

### PR DESCRIPTION
**Pull request by ocket8888**
#### What does this PR do?

Fixes some bugs with error handling in the `compare` tool and `genConfigRoutes.py` script, and moved route generation for CDN snapshots to be entirely handled by the route generation script, as well as adding the actual `/api/1.3/servers/{{server}}/configfiles/ats` route to the generated list.

#### Which TC components are affected by this PR?

- [ ] Documentation
- [ ] Grove
- [ ] Traffic Analytics
- [ ] Traffic Monitor
- [ ] Traffic Ops
- [ ] Traffic Ops ORT
- [ ] Traffic Portal
- [ ] Traffic Router
- [ ] Traffic Stats
- [ ] Traffic Vault
- [x] Other - The `compare` tool suite

#### What is the best way to verify this PR?
Build `traffic_ops/testing/compare/compare.go`, run the `traffic_ops/testing/compare/genConfigRoutes.py` script and pass the output to the compiled `compare` binary. Verify results. For more information, consult [the documentation](https://traffic-control-cdn.readthedocs.io/en/latest/tools/compare.html).

#### Check all that apply

- [ ] This PR includes tests
- [ ] This PR includes documentation updates
- [ ] This PR includes an update to CHANGELOG.md
- [ ] This PR includes all required license headers
- [ ] This PR includes a database migration (ensure that migration sequence is correct)
- [ ] This PR fixes a serious security flaw. Read more: [www.apache.org/security](http://www.apache.org/security/)

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2778/
Test FAILed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2779/
Test FAILed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2780/
Test FAILed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2782/
Test FAILed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2783/
Test FAILed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2784/
Test FAILed.

File: traffic_ops/testing/compare/compare.go [hunk 1]
```diff
@@ -239,46 +257,48 @@ func (to *Connect) get(route string) (string, error) {
 	if err != nil {
 		return "", err
 	}
-	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("Accept", "application/json")
+
+	// Should wait for any retries to complete before sending a request
+	to.mutex.Lock()
```
Comment (dangogh) [inline traffic_ops/testing/compare/compare.go]: rather than unlocking before each return,  better to `defer to.mutex.Unlock()` here..

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2786/
Test FAILed.

**Pull request merged**